cni/network/network_windows_test.go

```diff
@@ -301,3 +301,59 @@ func TestDSRPolciy(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNetworkNameFromCNS(t *testing.T) {
+	plugin, _ := cni.NewPlugin("name", "0.3.0")
+	tests := []struct {
+		name          string
+		plugin        *NetPlugin
+		netNs         string
+		nwCfg         *cni.NetworkConfig
+		ipamAddResult *IPAMAddResult
+		want          string
+	}{
+		{
+			name: "Runtime network polices",
+			plugin: &NetPlugin{
+				Plugin:      plugin,
+				nm:          network.NewMockNetworkmanager(),
+				ipamInvoker: NewMockIpamInvoker(false, false, false),
+				report:      &telemetry.CNIReport{},
+				tb:          &telemetry.TelemetryBuffer{},
+			},
+			netNs: "net",
+			nwCfg: &cni.NetworkConfig{
+				CNIVersion:   "0.3.0",
+				Name:         "azure",
+				MultiTenancy: true,
+			},
+			ipamAddResult: &IPAMAddResult{
+				ncResponse: &cns.GetNetworkContainerResponse{
+					MultiTenancyInfo: cns.MultiTenancyInfo{
+						ID: 1,
+					},
+				},
+				ipv4Result: &cniTypesCurr.Result{
+					IPs: []*cniTypesCurr.IPConfig{
+						{
+							Address: net.IPNet{
+								IP:   net.ParseIP("10.240.0.5"),
+								Mask: net.CIDRMask(24, 32),
+							},
+						},
+					},
+				},
+			},
+			want: "azure-vlan1-10-240-0-0_24",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			networkName, err := tt.plugin.getNetworkName(tt.netNs, tt.ipamAddResult, tt.nwCfg)
```

Comment:
Do we want to cover the error case scenario as well ? Since this test case is checking the happy path right now.

Reply:
At this point we only wanted to make sure that we're producing correct CIDR when the CNS response is valid. Seems that the failure path is not that critical for unit test.

Reply:
Checking the failure cases is absolutely critical in unit tests, come on. You should at least be testing that the input validation that you added correctly returns the error that you added when the input is invalid. How else do you know it rejects bad inputs?

Reply:
I added the unit test that was asked for this particular case. Yes, in general unit test is critical for non-happy paths. I didn't consider it critical here, since I was informed malformed CIDR is not reaching to this function. Nevertheless, I will add a test case for malformed CIDR since it is the only error check that I have added.